webbpsf/optics.py

```diff
@@ -465,13 +465,18 @@ def __init__(self, name='GR700XD', which='Bach',
     def get_opd(self, wave):
         """ Make an OPD array corresponding to the cylindrical weak lens
         used for defocusing the spectrum in the perpendicular-to-dispersion direction.
+        
+        Parameters
+        ----------
+        wave : float or obj
+            either a scalar wavelength (meters) or a Wavefront object
         """
 
         if isinstance(wave, poppy.Wavefront):
             wavelength = wave.wavelength
         else:
-            wavelength = float(wave)
-            wave = poppy.Wavefront(wavelength=wave)
+            wave = poppy.Wavefront(wavelength=float(wave))
+            wavelength = wave.wavelength
```

Comment:
This seems a minor but benign change; I'm curious if there was any subtle bug or issue with the prior code? Or if not then is this just a stylistic change for clarity?

Reply:
During development, I ran into an issue with type definitions. For instance, `wavelength` is actually expected to be an astropy unit in many places (particularly print and debug statments), so the change that I implemented always forces this by explicitly setting `wavelength = wave.wavelength`. I also wanted to make this consistent throughout the document, so that's why you see the change multiple times.